### PR TITLE
Add Saved Question button header and new, empty details sidebar

### PR DESCRIPTION
**Pull request by daltojohnso**
Note: plz ignore broken cypress tests (this is merging into a feature branch) as I intentionally removed functionality that will eventually live in the new sidebar 🙂 -- that functionality being the pencil icon dropdown and the tooltip for showing a question's `description`.

Current appearance of the new header button + sidebar: 

Screen Shot 2021-04-22 at 3 57 16 PM

mouse hover or sidebar open:
Screen Shot 2021-04-22 at 3 57 24 PM

current sidebar appearance
Screen Shot 2021-04-22 at 3 57 33 PM

### CODE REVIEW

File: frontend/src/metabase/lib/moderation.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function getModerationStatusIcon(status) {
+  switch (status) {
+    case "verification":
+      return "verified";
+    case "flag":
+      return "warning_colorized";
+    case "question":
+      return "clarification";
+  }
+}
```
Comment (kulyk) [inline frontend/src/metabase/lib/moderation.js:10]: nip: I usually find objects pretty handy in cases like that and they add more declarative flavor to the code. 

Example:

```js
const statusIconMap = {
   verification: "verified",
   flag: "warning_colorized",
   question: "clarification",
}
return statusIconMap[status];
```

File: frontend/src/metabase/query_builder/components/view/ViewHeader.jsx [hunk 2]
```diff
@@ -104,23 +111,17 @@ export class ViewTitleHeader extends React.Component {
...
+            <SavedQuestionHeaderButtonContainer>
+              <SavedQuestionHeaderButton
                 question={question}
-                onOpenModal={onOpenModal}
+                active={isShowingQuestionDetailsSidebar}
+                onClick={() =>
+                  isShowingQuestionDetailsSidebar
+                    ? onCloseQuestionDetails()
+                    : onOpenQuestionDetails()
+                }
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/view/ViewHeader.jsx:122]: nip: we can just pass a function reference and avoid creating a new function causing extra re-renderings

```js
onClick={
  isShowingQuestionDetailsSidebar
    ? onCloseQuestionDetails
    : onOpenQuestionDetails
}
```

File: frontend/src/metabase/query_builder/components/view/sidebars/QuestionDetailsSidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import SidebarContent from "metabase/query_builder/components/SidebarContent";
+
+function QuestionDetailsSidebar() {
+  return (
+    <SidebarContent className="full-height px1">
+      question details sidebar
+    </SidebarContent>
+  );
+}
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/view/sidebars/QuestionDetailsSidebar.jsx:10]: Not sure why it contains just "question details sidebar"? Or is it a WIP?

**Review by kulyk — APPROVED** [review]
Looks good!

File: frontend/src/metabase/query_builder/components/view/ViewHeader.jsx (hunk 2 above)
Comment (daltojohnso) [reply frontend/src/metabase/query_builder/components/view/ViewHeader.jsx]: good catch

File: frontend/src/metabase/lib/moderation.js (hunk 1 above)
Comment (daltojohnso) [reply frontend/src/metabase/lib/moderation.js]: nice, I'll take it a step further to skip having to name a variable

File: frontend/src/metabase/query_builder/components/view/sidebars/QuestionDetailsSidebar.jsx (hunk 3 above)
Comment (daltojohnso) [reply frontend/src/metabase/query_builder/components/view/sidebars/QuestionDetailsSidebar.jsx:10]: ya, this is a WIP, that's just random text for no real reason at all. lots of changes and I didn't want a huge PR--my next 10-20 PRs in this feature branch will handle filling it out to look like this:

Screen Shot 2021-04-23 at 8 33 11 AM

**Review by kulyk — APPROVED** [review]

**Pull request merged**